src/main/java/org/apache/sling/graphql/core/engine/DefaultQueryExecutor.java

```diff
@@ -147,8 +147,8 @@ public ValidationResult validate(@NotNull String query, @NotNull Map<String, Obj
                         Arrays.toString(selectors)));
             }
             LOGGER.debug("Resource {} maps to GQL schema {}", queryResource.getPath(), schemaDef);
-            final TypeDefinitionRegistry typeDefinitionRegistry = getTypeDefinitionRegistry(schemaDef, queryResource, selectors);
-            final GraphQLSchema schema = buildSchema(typeDefinitionRegistry, queryResource);
+            CurrentThreadResource.setCurrentResource(queryResource);
+            final GraphQLSchema schema = getSchema(schemaDef, new CurrentThreadResource(), selectors);
```

Comment:
I don't think that this design is clear. I'd prefer if you didn't have a static method for setting the "proxied" resource, but rather passed it into the constructor.